docs/examples.rst

```diff
@@ -4,4 +4,391 @@
+angle.
+
+   .. list-table::
+      :widths: 50 50
+      :align: center
+
+      * -
+
+         .. do not delete this table row
+
+        -
+
+      * -
+
+         .. image:: _static/pendulum.gif
+            :alt: Inverted Pendulum Simulation
+
+         |nbsp| |nbsp| |nbsp| |nbsp| Adaptive Controller
+
+        -
+
+         .. image:: _static/pendulum_no-learn.gif
+            :alt: Inverted Pendulum Adaptive Simulation
+
+         |nbsp| |nbsp| Non-adaptive Controller
+
+You can also try moving the **Extra Mass** slider up or down in the GUI and
+observe how the added or reduced mass affect the position of the pendulum,
+especially as it approaches the horizontal position where the torque due to
+gravity is at its peak.
+
+Reinforcement Learning
+----------------------
+
+This example demonstrates how an adaptive FPGA ensemble can be used to
+implement an agent capable of reinforcement learning. The agent can perform
+three actions (go forward, rotate clockwise, rotate counter-clockwise), and
+has a "radar" that provides it the distance to any obstacles directly ahead,
+directly to its sides, and on the forward diagonals. The agent is controlled
+by a basal ganglia (BG) action selection system that determines which of the
+three actions the agent is to perform, based on the input utility values to the
+basal ganglia.
+
+The radar data is fed into the adaptive FPGA ensemble to compute the BG
+utility values. When the simulation starts, the BG utility values start at
+0.8 for moving forward, and 0.7 and 0.6 for rotating counter-clockwise and
+clockwise respectively. Thus, if the utility values are not modified,
+the agent should be biased to move in a forward direction.
+
+A reward system is set up to give a positive value when moving forward, and
+no reward if the agent is stationary or has crashed into a wall.  In theory,
+this should cause it to learn to avoid obstacles.  In particular, it will
+start to turn before it hits the obstacle, as a continual forward movement
+(albeit at an angle) will provide it with a higher reward.
+
+From the reward system, an error signal is computed. The error signal in this
+case is very simple: the difference between the computed utility and the
+instantaneous reward.  This error signal should only be applied to whatever
+action is currently being chosen (although it isn't quite perfect at doing
+this).  Note that this means it cannot learn to do actions that will lead to
+*future* rewards.
+
+The error signal is fed into the adaptive FPGA ensemble to modulate it's,
+which in turn modifies the mapping between the different radar states and the
+learned action. In this way, the agent learns to associate specific radar
+states with the action that will provide it the highest reward (and hopefully
+avoid the walls).
+
+This example has an optional requirement.
+
+.. rst-class:: compact
+
+1. (Optional) ``keyboard-state`` branch of Nengo GUI. This will allow you to

```

Comment:
(no action required) Terry mentioned he plans to merge in this branch soon, let's keep an eye on that and remember update this section when it's merged. (now at least two of us will have it in mind :+1:)